### ISSUE TITLE
Incomplete closure checking and covers

### ISSUE THREAD

**Issue by jcrozum** [CASCI-lab/schematodes#3]
Here is a set that is giving some trouble:

```
[
  [2, 1, 0],
  [0, 2, 1],
  [1, 2, 0],
  [0, 1, 2],
]
```

We should get [012, 12], [120, 01], [021, 02], and [210, 02]. Previously, we were getting some extra symmetries that did not exist. This has been fixed [here](https://github.com/CASCI-lab/schematodes/tree/improper-closure-check-fix), but we still miss one of the schema.

**Comment by jcrozum** [comment]
I just added branch [here](https://github.com/CASCI-lab/schematodes/tree/alternate-approach) that fixes this issue completely. Digging into this issue has revealed that the same-symbol symmetry is more complicated than we initially thought. I think it can be more properly handled by wrapping the block on line 177 of lib.rs in an if-statement so that the following code only runs when we want to exclude same-symbol symmetry:
```rust
// we skip the swaps that include trivial columns because these are not faithful group actions
if swaps.iter().any(|swap| {
    swap.iter()
        .any(|x| trivial_redescription_columns.contains(x))
}) {
    continue;
}
```
But I'm not sure whether or not that would actually give the correct results in all cases. As a simple example to illustrate the difficulty, consider
```
0122
1022
2012
0212
2102
1202
```
Here, maximal "bubbly" symmetry group is $S_{0,1,2}$ acting on the whole set. We also could have $S_{0,1}\times S_{2,3}$ acting on the first two rows, which is also an inclusion-maximal symmetry group (though it is not faithful). We would miss this if we tried to do the same-symbol symmetry after-the-fact.

**Comment by jcrozum** [comment]
@austin-marcus I don't want to jinx it, but I think I fixed the problem I mentioned to you earlier, so you can rest easy. I'm tagging you here in case you're curious about the rule that caused the problems. We can discuss in more detail some other time.

**Comment by jcrozum** [comment]
The branch I mentioned is now merged in #4.
